File: cpp/2/main.cpp
#include <iostream>
#include <string>
#include <stdexcept>

#include "arabrzym.hpp"

using namespace std;

int main(int argc, char* argv[]) {
    for (int i = 1; i < argc; i++) {
        try {
            try {
                int arab = stoi(argv[i]);
                cout << ArabRzym::arab2rzym(arab) << endl;
            } catch (const invalid_argument& error) {
                cout << ArabRzym::rzym2arab(argv[i]) << endl;
            }
        } catch (const ArabRzymException& error) {
            cout << error.what() << endl;
        }
    }
    return 0;
}

File: java/2/Main.java

class Main{
    public static void main(String[] args){
        for (int i = 0; i < args.length; i++){
            try{
                try{
                    System.out.println(ArabRzym.arab2rzym(Integer.parseInt(args[i])));
                }
                catch(NumberFormatException error){
                    System.out.println(ArabRzym.rzym2arab(args[i]));
                }
            }
            catch(ArabRzymException error){
                System.out.println(error.getMessage());
            }  
        }
    }
}